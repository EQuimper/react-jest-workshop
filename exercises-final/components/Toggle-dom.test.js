import React from 'react'
import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import Toggle from './Toggle'

test('invokes the onToggle prop when clicked', () => {
  const onToggle = jest.fn()
  const wrapper = mountToggle({onToggle})
  clickButton(wrapper)
  expect(onToggle).toHaveBeenCalledTimes(1)
  expect(onToggle).toBeCalledWith(true)
})

test('changes the class to toggle--on when clicked', () => {
  const wrapper = mountToggle()
  clickButton(wrapper)
  expect(mountToJson(wrapper)).toMatchSnapshot()
})

/**
 * Uses enzyme to mount the Toggle component
 * @param {Object} props - the props to mount the component with
 * @return {Object} - the enzyme wrapper
 */
function mountToggle(props = {}) {
  return mount(
    <Toggle onToggle={() => {}} {...props}>Toggle Me</Toggle>
  )
}

/**
 * finds the button in the given wrapper and simulates a click event
 * @param {Object} wrapper - the enzyme wrapper
 */
function clickButton(wrapper) {
  wrapper.find('button').first().simulate('click')
}
