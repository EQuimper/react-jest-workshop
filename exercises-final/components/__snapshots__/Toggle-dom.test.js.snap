exports[`test changes the class to toggle--on when clicked 1`] = `
<Toggle
  onToggle={[Function onToggle]}>
  <div
    className="toggle toggle--on">
    <button
      onClick={[Function anonymous]} />
  </div>
</Toggle>
`;
